lib/_http_client.js

```diff
@@ -197,6 +197,9 @@ function ClientRequest(input, options, cb) {
   }
   this.insecureHTTPParser = insecureHTTPParser;
 
+  options.path = options.path && method === 'CONNECT' &&
```

Comment:
Should `options.path` be required when the method is `CONNECT`?

Reply:
Two things:

1. Don't modify `options.path` - the `options` object can be owned by the caller, not by this method; mutations will be visible.

2. `CONNECT` requires a `<host>:<port>` combo. Chopping off the leading slash is the first step but it should also check it's a valid combo.

I don't think we have a utility function for that right now and checking if ```new URL(`http://${options.path}`)``` throws isn't robust (accepts e.g. `evil.com:666/good.org:777`.)

I think the regex would look something like this...
```js
^[\-_0-9A-Za-z]+(\.[\-_0-9A-Za-z]+)*\.?:([1-9]|[1-9][0-9]{1,2}|[1-5][0-9]{3}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$
```
(I hope I'm not introducing catastrophic backtracing here...)

It should match inputs like `example.com:80` but also `example.com.:80` (FQDN), `example:80`, `example.:80`, etc., but not e.g. `example.com:0` or `example.com:65536`.

Reply:
> Should `options.path` be required when the method is `CONNECT`?

@szmarczak 
Apparently yes, there was a test in which `options.path` is undefined so `charAt` throws an error.

Reply:
> Don't modify options.path - the options object can be owned by the caller, not by this method; mutations will be visible.

Okay, I'll try avoiding that.

> I don't think we have a utility function for that right now and checking if new URL(`http://${options.path}`) throws isn't robust (accepts e.g. evil.com:666/good.org:777.)

Should we include this check as a utility function in `url.js` or must check for a valid `<host>:<port>` here in `_http_client.js`?
Also it would require some additional tests right?

Reply:
It certainly shouldn't be exposed as a public function, if that is what you're asking.

An internal function in `_http_client.js` is okay but if you add it to `lib/internal/http.js`, you can unit-test it if you add a `// Flags: --expose-internals` magic comment to the test file.